ContactList/ContactList/App.cs

```diff
@@ -0,0 +1,198 @@
+                            string deleteSurname = Console.ReadLine();
+
+                            if (string.IsNullOrEmpty(deleteName) || string.IsNullOrEmpty(deleteSurname))
+                            {
+                                throw new InvalidInputDataException("Invalid input data, try again.");
+                            }
+
+                            _contactBookServices.DeleteContact(deleteName, deleteSurname);
+                            Console.WriteLine($"Contact {deleteName},{deleteSurname} was deleted");
+                            break;
+
+                        case 3:
+                            try
+                            {
+                                Console.WriteLine("Input the paramiter for search contact:");
+
+                                switch (Console.ReadLine())
+                                {
+                                    case "Name":
+                                        Console.WriteLine("Input name:");
+                                        string searchName = Console.ReadLine();
+
+                                        if (string.IsNullOrEmpty(searchName))
+                                        {
+                                            throw new InvalidInputDataException("Inputed name is not possible to be null.");
+                                        }
+                                        var searchNameContacts = _contactBookServices.SearchContactAsync("Name", searchName);
+
+                                        foreach (Contact contact in searchNameContacts)
+                                        {
+                                            Console.WriteLine(contact.Name + ' ' + contact.Surname + ' ' + contact.Number);
+                                        }
+                                        break;
+
+                                    case "Surname":
+                                        Console.WriteLine("Input surname:");
+                                        string searchSurname = Console.ReadLine();
+
+                                        if (string.IsNullOrEmpty(searchSurname))
+                                        {
+                                            throw new InvalidInputDataException("Inputed surname is not possible to be null.");
+                                        }
+                                        var searchSurnameContacts = _contactBookServices.SearchContactAsync("Surname", searchSurname);
+
+                                        foreach (Contact contact in searchSurnameContacts)
+                                        {
+                                            Console.WriteLine(contact.Name + ' ' + contact.Surname + ' ' + contact.Number);
+                                        }
+                                        break;
+
+                                    case "Number":
+                                        Console.WriteLine("Input number:");
```

Comment:
Most of the code for search by name/surname/number is a duplicate, should have moved to the common method